libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -749,8 +757,13 @@ void AP_DDS_Client::on_request(uxrSession* uxr_session, uxrObjectId object_id, u
             break;
         }
 
-        GCS_SEND_TEXT(MAV_SEVERITY_INFO, "%s Request for %sing received", msg_prefix, arm_motors_request.arm ? "arm" : "disarm");
-        arm_motors_response.result = arm_motors_request.arm ? AP::arming().arm(AP_Arming::Method::DDS) : AP::arming().disarm(AP_Arming::Method::DDS);
+        if (external_control->is_enabled()) {
```

Comment:
Can we instead put an `arm` function into the external control library, and check `is_enabled` there, then call `AP::arming().arm` there?

Reply:
same for the other services though, right?

Reply:
Yep, I think so. For the over services, that could/should be it's own PR. I'll do that now.

Reply:
so you would call `external_control->arm()`, which would fail if either the external control is disabled or arming failed.

Reply:
yes

Reply:
Here's the implementation. Once this merges, rebase. 
https://github.com/ArduPilot/ardupilot/pull/28609

Reply:
Please rebase and fold the arming in.

Reply:
done and pushed